### PR TITLE
COmanage_plugins_role

### PR DESCRIPTION
**Pull request by ioigoume**
Ansible Role for COmanage Registry Plugins

### CODE REVIEW

File: roles/comanage-registry-plugins/vars/Debian.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+---
+apache_user: "www-data"
```
Comment (NicolasLiampotis) [inline roles/comanage-registry-plugins/vars/Debian.yml:2]:
```
comanage_registry_plugins_default_webuser: "www-data"
```

File: roles/comanage-registry-plugins/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+- name: Include OS-specific variables
+  include_vars: "{{ item }}"
+  with_first_found:
+    - "{{ ansible_distribution }}-{{ ansible_distribution_release}}.yml"
+    - "{{ ansible_distribution }}-{{ ansible_distribution_major_version}}.yml"
+    - "{{ ansible_distribution }}.yml"
+    - "{{ ansible_os_family }}.yml"
+  tags:
+  - always
+
```
Comment (NicolasLiampotis) [inline roles/comanage-registry-plugins/tasks/main.yml:12]: Add:
```
- name: Define COmanage Registry Plugins web user  
   set_fact:
     comanage_registry_plugins_webuser: "{{ comanage_registry_plugins_default_webuser }}"  
   when: comanage_registry_plugins_webserver is not defined  
   tags:
     - always
```

File: roles/comanage-registry-plugins/tasks/preinstall.yml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+- name: Install git if it is not present
+  apt:
+    name: git
+    state: present
+    update_cache: yes
+  become: yes
+
+- name: Clear cache of COmanage
+  command: su {{ apache_user }} -s /bin/sh -c ./Console/clearcache
```
Comment (NicolasLiampotis) [inline roles/comanage-registry-plugins/tasks/preinstall.yml:11]: apache_user --> comanage_registry_plugins_webuser

**Review by NicolasLiampotis — CHANGES_REQUESTED** [review]

File: roles/comanage-registry-plugins/tasks/install.yml [hunk 4]
```diff
@@ -0,0 +1,26 @@
+---
+
+- name: Ensure local/Plugins directory exists
```
[the remark this answers is not in the document]
Comment (ioigoume) [reply roles/comanage-registry-plugins/tasks/install.yml:3]: Done

File: roles/comanage-registry-plugins/defaults/main.yml [hunk 5]
```diff
@@ -0,0 +1,26 @@
+---
+
+comanage_registry_plugins:
```
[the remark this answers is not in the document]
Comment (ioigoume) [reply roles/comanage-registry-plugins/defaults/main.yml:3]: Done

File: roles/comanage-registry-plugins/tasks/install.yml [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+- name: Ensure {{ item.name }} is cloned under local/Plugins directory
+  git:
+    repo: "{{ item.repository }}"
+    dest: "{{ comanage_path }}/local/Plugin/{{ item.name }}"
+    version: "{{ item.version }}"
+    accept_hostkey: yes
+  become: yes
+
+- name: Create Schema
```
[the remark this answers is not in the document]
Comment (ioigoume) [reply roles/comanage-registry-plugins/tasks/install.yml:19]: Done

File: roles/comanage-registry-plugins/vars/Debian.yml (hunk 1 above)
Comment (ioigoume) [reply roles/comanage-registry-plugins/vars/Debian.yml:2]: Done

File: roles/comanage-registry-plugins/tasks/preinstall.yml (hunk 3 above)
Comment (ioigoume) [reply roles/comanage-registry-plugins/tasks/preinstall.yml:11]: Done

File: roles/comanage-registry-plugins/tasks/configure.yml [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+  cron:
+    name: "{{ item.name }}"
+    minute: "{{ item.minute }}"
+    job: "{{ item.job }}"
+    user: "{{ item.user }}"
+    cron_file: "{{ item.filename }}"
+    backup: true
+  loop: "{{ cron_jobs | default([]) }}"
+  become: true
+  when: cron_jobs is defined
```
[the remark this answers is not in the document]
Comment (ioigoume) [reply roles/comanage-registry-plugins/tasks/configure.yml]: Done

File: roles/comanage-registry-plugins/tasks/main.yml [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+# Include pre-installation tasks
+- name: Include pre-Install tasks
+  include_tasks:
+    file: preinstall.yml
+    apply:
+      tags:
+        - preinstall
```
[the remark this answers is not in the document]
Comment (ioigoume) [reply roles/comanage-registry-plugins/tasks/main.yml:26]: Leaving as is

File: roles/comanage-registry-plugins/tasks/main.yml [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+  tags:
+    - preinstall
+
+# Include installation tasks
+- name: Include Installation tasks
+  include_tasks:
+    file: install.yml
+    apply:
+      tags:
+        - install
```
[the remark this answers is not in the document]
Comment (ioigoume) [reply roles/comanage-registry-plugins/tasks/main.yml:36]: Leaving as is

File: roles/comanage-registry-plugins/tasks/main.yml [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+  tags:
+  - install
+
+# Include configuration tasks
+- name: Include Configure tasks
+  include_tasks:
+    file: configure.yml
+    apply:
+      tags:
+        - configure
```
[the remark this answers is not in the document]
Comment (ioigoume) [reply roles/comanage-registry-plugins/tasks/main.yml:47]: Leaving as is

**Pull request merged**